proxy/src/telemetry/metrics/mod.rs

```diff
@@ -68,19 +68,21 @@ struct Metrics {
     response_total: Metric<Counter, Arc<ResponseLabels>>,
     response_latency: Metric<Histogram, Arc<ResponseLabels>>,
 
-    tcp_accept_open_total: Metric<Counter, Arc<TransportLabels>>,
-    tcp_accept_close_total: Metric<Counter, Arc<TransportCloseLabels>>,
+    tcp: TcpMetrics,
 
-    tcp_connect_open_total: Metric<Counter, Arc<TransportLabels>>,
-    tcp_connect_close_total: Metric<Counter, Arc<TransportCloseLabels>>,
+    start_time: u64,
+}
 
-    tcp_connection_duration: Metric<Histogram, Arc<TransportCloseLabels>>,
-    tcp_connections_open: Metric<Gauge, Arc<TransportLabels>>,
+#[derive(Debug, Clone)]
+struct TcpMetrics {
```

Comment:
I was considering making this change for a while, but I wasn't sure whether or not it actually makes the code clearer...actually seeing it, I think it was a good call.

Reply:
Idle thought: I wonder if locking the TCP and protocol-level metrics separately would lead to better or worse performance...we could even look into sending the transport-level and protocol-level events to totally separate aggregate tasks? I don't have an intuitive sense for whether or not this would actually improve performance though.